### PR TITLE
CC-94: Optimize inputs for mobile users

### PR DESCRIPTION
**Pull request by vishesh-ats**
- resolves https://github.com/Components-care/backlog/issues/94

### CODE REVIEW

File: src/standalone/UIKit/TextFieldWithHelp.tsx [hunk 1]
```diff
@@ -1,51 +1,94 @@
...
+		important,
+		valueFormatted,
+		onClear,
+		onChange,
+		...muiProps
+	} = props;
 	const inputClasses = useInputStyles({ important });
+	const [value, setValue] = useState(muiProps.value);
+
+	const isTouchDevice = useCallback(() => {
```
Comment (TQjona) [inline src/standalone/UIKit/TextFieldWithHelp.tsx:44]: turn this into a util

**Review by TQjona — CHANGES_REQUESTED** [review]
Why did you modify Integer/Decimal/Currency Inputs? Why did you make them all stateful? They shouldn't be stateful.

**Comment by vishesh-ats** [comment]
> Why did you modify Integer/Decimal/Currency Inputs? Why did you make them all stateful? They shouldn't be stateful.
- We are displaying formatted value for `Integer/Decimal/Currency` Inputs. But now we are managing value in `TextFieldWithHelp` component, so we are loosing formatting. To manage the formatted value, adjusted those changes.

File: src/standalone/UIKit/TextFieldWithHelp.tsx (hunk 1 above)
Comment (vishesh-ats) [reply src/standalone/UIKit/TextFieldWithHelp.tsx]: @TQjona Updated changes.. Please Review..

**Comment by TQjona** [comment]
And in doing so you break the form engine. The only thing you need to do is clear the input value in the text field and trigger an onChange event if the user clicks the clear button => https://stackoverflow.com/a/46012210

**Comment by TQjona** [comment]
And if that doesn't work add a new property to handle the clearing of the value.

File: src/utils/isTouchDevice.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ */
+const isTouchDevice = (): boolean => {
+	return (
+		"ontouchstart" in window ||
+		navigator.maxTouchPoints > 0 ||
+		navigator.msMaxTouchPoints > 0
+	);
+};
+
+export default isTouchDevice;
```
Comment (TQjona) [inline src/utils/isTouchDevice.ts:12]: missing re-export in src/utils/index.ts

Comment (vishesh-ats) [reply src/utils/isTouchDevice.ts:12]: @TQjona update it..

**Pull request closed without merging**